src/commonMain/kotlin/uk/gov/hmrc/calculator/Calculator.kt

```diff
@@ -367,10 +375,19 @@ class Calculator @JvmOverloads constructor(
         return amount
     }
 
-    private fun shouldApplyStandardTapering(yearlyWageAfterPension: Double) = taxCodeType is StandardTaxCode &&
-        !userSuppliedTaxCode && yearlyWageAfterPension.shouldApplyTapering()
-
     private val taxCodeType: TaxCode by lazy {
         this.taxCode.toTaxCode()
     }
+
+    data class StudentLoanPlans(
+        val hasPlanOne: Boolean = false,
+        val hasPlanTwo: Boolean = false,
+        val hasPlanFour: Boolean = false,
+        val hasPostgraduatePlan: Boolean = false,
+    )
+
+    data class PensionContribution(
+        val method: PensionMethod? = null,
+        val contributionAmount: Double? = null,
```

Comment:
Is it correct that these are both nullable? Wouldn't it just be the object wasn't created if either were null, as both would be required to calculate a pension so you either want both present, or neither.
```suggestion
        val method: PensionMethod,
        val contributionAmount: Double,
```